controllers/etcd/reconciler.go

```diff
@@ -400,12 +400,10 @@ func isPeerTLSChangedToEnabled(peerTLSEnabledStatusFromMembers bool, configMapVa
 
 func bootstrapReset(etcd *druidv1alpha1.Etcd) {
 	etcd.Status.Members = nil
-	etcd.Status.ClusterSize = pointer.Int32(etcd.Spec.Replicas)
 }
 
 func clusterInBootstrap(etcd *druidv1alpha1.Etcd) bool {
-	return etcd.Status.Replicas == 0 ||
-		(etcd.Spec.Replicas > 1 && etcd.Status.Replicas == 1)
+	return etcd.Spec.Replicas >= 1 && etcd.Status.Replicas == 0
```

Comment:
`func clusterInBootstrap(){...}` should return true if it's the bootstrap case else false. But I found one scenario, in that scenario it is returning false.
Let's take the scenario:
Suppose we are bootstrapping a HA cluster i.e 0->3 and only 1 etcd pod come-up and 2 fails to join may be due to volume attachment issue.
`etcd.Spec.Replicas` is 3 and as one members joins then `etcd.Status.Replicas` should become `1` then your condition will returns false but it is still a bootstrap case.

Reply:
But the current logic will return true even during cluster scale-up from 1 to 3, right? @unmarshall does it make sense to use `return etcd.Spec.Replicas >= 1 && etcd.Status.ReadyReplicas == 0` instead, because the case that Ishan mentioned is quite possible, so even during bootstrap from 0->3, when 1 replica is up and two others are still coming up, the replica will not be ready, hence `ReadyReplicas` will still be `0`, until the cluster is fully formed and ready.

Reply:
Consider another case:
Initial state: 1 member (healthy)
Scale up is triggered from 1 -> 3. Now 2 additional pods come up (these are not yet ready).
Now the first pod goes down and post re-start it is unable to come up. So now the `etcd.Spec.Repliacs >=1` and `etcd.Status.ReadyReplicas == 0` however this is a scale up case.

Reply:
Looks like this particular problem might be solved only if druid knows which of the etcd members have already been initialized and part of the cluster. Maybe then we might need another field such as `JoinedReplicas` that holds information about the replicas that have already joined the cluster, and this can be as simple as a string array with non-repeating pod names. Essentially, druid, ie custodian controller, needs to watch for pod readiness of the etcd pods, and as soon as any pod gets ready, it must add it to this particular list, hence maintaining information about which pods have already joined the cluster. So druid etcd controller will simply consult this list when deciding whether it's a scale-up case or bootstrapping case. But again, we don't know how this plays out with quorum loss scenarios.

Anyway, all this complexity is completely avoided when we move to harmonized scaling, because then we would never have a bootstrap case in the first place, so all this complex implementation would be redundant.

Reply:
Post our discussion today amongst (Shreyas, Aaron, Ishan, Abhishek and me) we think that it might be wise to just remove the `clusterInBootstrap` check.
Reason:
There is no fool proof way to determine if the cluster is in bootstrap. Today the cluster can be in bootstrap in 2 possible cases
- It is a fresh cluster. In this case initially there will not be any STS till etcd-controller reconciles.
- Cluster is woken up from hibernation - in this case STS exists, its replicas are scaled from 0-> <etcd.Spec.Replicas>

Also the only action taken if the cluster is in bootstrap is that it sets `etcd.Status.Members` to nil. We think that this can be removed since custodian controller periodically will recalculate the list of registered members and replaces the existing slice of Members with a new slice. Therefore eventually consistency will be there.

I will wait for a day and if no one has any further objections then i will remove `clusterInBootstrap` and `bootstrapReset` functions.

Reply:
After giving it some thought, and I would vote to have `clusterInBootstrap` and `bootstrapReset` removed, simply because these will anyway go away when #589 comes in, and adding any new fields to Etcd status would simply be redundant because those will no longer be needed later and will have to be eventually removed.

Reply:
I have made the changes as discussed. `clusterInBootstrap` and `bootstrapReset` methods and their usages have been now removed.